src/Transloadit.ts

```diff
@@ -114,14 +60,14 @@ function getHrTimeMs(): number {
 
 function checkResult<T>(result: T | { error: string }): asserts result is T {
   // In case server returned a successful HTTP status code, but an `error` in the JSON object
-  // This happens sometimes when createAssembly with an invalid file (IMPORT_FILE_ERROR)
+  // This happens sometimes, for example when createAssembly with an invalid file (IMPORT_FILE_ERROR)
   if (
     typeof result === 'object' &&
     result !== null &&
     'error' in result &&
     typeof result.error === 'string'
   ) {
-    throw makeTransloaditError(new Error('Error in response'), result)
+    throw new ApiError({ body: result }) // in this case there is no `cause` because we don't have a HTTPError
```

Comment:
Nitpicky grammar issue:

```suggestion
    throw new ApiError({ body: result }) // in this case there is no `cause` because we don't have an HTTPError
```